misk-hibernate/src/main/kotlin/misk/hibernate/ReflectionQueryFactory.kt

```diff
@@ -127,13 +276,14 @@ internal class ReflectionQuery<T : DbEntity<T>>(
     val typedQuery = session.hibernateSession.createQuery(query)
     return traceSelect {
       session.disableChecks(disabledChecks) {
-        typedQuery.getSingleResult()
+        typedQuery.singleResult
       }
     }
   }
 
   private fun effectiveMaxRows(returnList: Boolean): Int {
     return when {
+      (maxRows == 1) -> 1 // If specifically setting maxRows to 1 leave it so we can use unique
```

Comment:
Discussed offline.

Reply:
We will trust the caller here. Add a test to prevent me from “fixing” this?

Reply:
Test exists